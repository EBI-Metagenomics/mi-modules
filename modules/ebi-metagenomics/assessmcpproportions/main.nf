
process ASSESSMCPPROPORTIONS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.1.5--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:0.1.5--pyhdfd78af_0' }"

    input:
    tuple val(meta), val(fwd_flag), val(rev_flag), path(fastq)

    output:
    tuple val(meta), path("*.tsv"), emit: tsv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def var_region = "${meta.var_region}"
    def strands = ""

    if (fwd_flag == "auto" && rev_flag == "auto") {
        strands = "FR"
    } else if (fwd_flag == "auto"){
        strands = "F"
    } else if (rev_flag == "auto") {
        strands = "R"
    }

    if (strands == ""){
        """
        touch ${prefix}_${var_region}_mcp_cons.tsv
        
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            mgnify-pipelines-toolkit: 0.1.5
        END_VERSIONS
        """
    } else{
        """
        assess_mcp_proportions \\
            -i ${fastq} \\
            -s ${prefix}_${var_region} \\
            -st ${strands} \\
            -o ./

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            mgnify-pipelines-toolkit: 0.1.5
        END_VERSIONS
        """
    }

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def var_region = "${meta.var_region}"

    """
    touch ${prefix}_${var_region}_mcp_cons.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: 0.1.5
    END_VERSIONS
    """
}
