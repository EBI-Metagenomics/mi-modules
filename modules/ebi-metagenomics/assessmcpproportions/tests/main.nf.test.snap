{
    "assessmcpproportions - FR": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,875c712aadf9b21bd535e555dbb4b7e1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,875c712aadf9b21bd535e555dbb4b7e1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:28:05.119433916"
    },
    "assessmcpproportions - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:31:40.203930947"
    },
    "assessmcpproportions - R": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,e4500d4e1aabab1e143063f4c26029ea"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,e4500d4e1aabab1e143063f4c26029ea"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:28:53.336604776"
    },
    "assessmcpproportions - FR - library_check": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,875c712aadf9b21bd535e555dbb4b7e1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "AMPLICON"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "AMPLICON"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,875c712aadf9b21bd535e555dbb4b7e1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:31:33.938320151"
    },
    "assessmcpproportions - F": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,1c566bc27d2bf777e76ffb8e52db7132"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,1c566bc27d2bf777e76ffb8e52db7132"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:28:27.577173142"
    },
    "assessmcpproportions - none": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "2": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ],
                "library_check_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        ""
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "var_region": "16S-V3-V4"
                        },
                        "test_16S-V3-V4_mcp_cons.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1991dfecb03dab8f53e250d7fe03f999"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T15:29:20.525923125"
    }
}